### PR TITLE
feat: generate dart code

### PR DESCRIPTION
**Pull request by dmccartney**
This generates and prepares a dart package for publication.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-npm test
-./dev/go/generate
+# TODO: cut pre-commit or uncomment before merging
+#npm test
+#./dev/go/generate
```
Comment (dmccartney) [inline .husky/pre-commit:6]: the pre-commit hook is really heavy -- I wonder if we need it?

**Review by nakajima — COMMENTED** [review]
Wouldn't say I'm qualified to review the dart stuff, but otherwise looks good to me!

**Review by mkobetic — APPROVED** [review]
I don't know anything about dart, but the overall structure looks good to me. Note that I'm proposing a small refactor in #24 which is likely to impact this, but I can fix that up if we decide to go with it.

**Review by elisealix22 — APPROVED** [review]
Looks good to me!

**Pull request merged**